fog/ledger/server/src/router_server.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+use futures::executor::block_on;
+use mc_common::logger::{log, Logger};
+use mc_fog_api::ledger_grpc;
+use mc_fog_ledger_enclave::LedgerEnclaveProxy;
+use mc_fog_uri::{ConnectionUri, FogLedgerUri, KeyImageStoreUri};
+use mc_util_grpc::{ConnectionUriGrpcioServer, ReadinessIndicator};
+use mc_util_uri::AdminUri;
+
+use crate::{
+    config::LedgerRouterConfig, router_admin_service::LedgerRouterAdminService,
+    router_service::LedgerRouterService,
+};
+
+pub struct LedgerRouterServer {
+    router_server: grpcio::Server,
+    admin_server: grpcio::Server,
+    client_listen_uri: FogLedgerUri,
+    admin_listen_uri: AdminUri,
+    logger: Logger,
+}
+
+impl LedgerRouterServer {
+    pub fn new<E>(
+        config: LedgerRouterConfig,
+        enclave: E,
+        shards: Arc<RwLock<HashMap<KeyImageStoreUri, Arc<ledger_grpc::KeyImageStoreApiClient>>>>,
+        logger: Logger,
+    ) -> LedgerRouterServer
+    where
+        E: LedgerEnclaveProxy,
+    {
+        let readiness_indicator = ReadinessIndicator::default();
+
+        let env = Arc::new(
+            grpcio::EnvBuilder::new()
+                .name_prefix("ledger-router-server".to_string())
+                .build(),
+        );
+
+        // Health check service - will be used in both router + admin interface
+        let health_service =
+            mc_util_grpc::HealthService::new(Some(readiness_indicator.into()), logger.clone())
+                .into_service();
+
+        // Build our router server.
+        // Init ledger router service.
+        let ledger_router_service = ledger_grpc::create_ledger_api(LedgerRouterService::new(
```

Comment:
Will the Ledger Router be able to service the unary API in addition to the streaming API? AFAIU, the code as it's currently written only supports one of the two, and I think it's the streaming API?

Reply:
That's correct. We'll be adding the unary API functionality as well as all the non-key-image functionality in a followup PR.